frontend/apps/filemanager/filemanagerbookinfo.lua

```diff
@@ -400,17 +402,28 @@ function BookInfo:setCustomMetadata(file, book_props, prop_key, prop_value)
         os.remove(custom_doc_settings.custom_metadata_file)
         DocSettings:removeSidecarDir(file, util.splitFilePathName(custom_doc_settings.custom_metadata_file))
     else
+        if book_props.pages then -- update a copy of original 'pages' if number of pages changed
+            local original_props = custom_doc_settings:readSetting("doc_props")
+            if original_props.pages ~= book_props.pages then
+                original_props.pages = book_props.pages
+                custom_doc_settings:saveSetting("doc_props", original_props)
+            end
```

Comment:
Do you propose to not use saveSetting when a setting is a table? Always?
(Many of my "Correct access to doc_settings" PRs should be reverted)
Like
https://github.com/koreader/koreader/blob/cc82ead981ed2bea5022f70d6b1218ed369377d5/frontend/apps/reader/readerui.lua#L475

Reply:
> Do you propose to not use saveSetting when a setting is a table? Always?

Uh, no, not taking such a responsability :)
May be it is fine and more readable the way you do it (even if in effect, it assigns the same thing, ie `a = a`).
Pinging @NiLuJe for thoughts.

Reply:
From what little I can see given than I haven't checked the context, you don't need the explicit saveSetting, as the reference hasn't changed, as @poir-z mentioned, so you cna just update the field like what he proposed?

Reply:
We all understand that. The question is which form we should favor ?
A: readSetting() and knowing it's a table and a shared reference, not needing :saveSetting() ?
B: readSetting() and saveSetting always so it's all explicite, whether it's a table (where it would be uneeded) or a number/string (where it is needed)

(Or no favoritism, and is it up to the writer ? :)

Reply:
Definitely A, the API returns native objects for a reason ;).

The read/save dance is only necessary when references need to be broken.

Reply:
(Barring extreme ambiguity or scoping concerns, i.e., as usual, common sense matters ;))